components/director/internal/domain/formationconstraint/constraint_operators_test.go

```diff
@@ -386,3 +385,249 @@ func TestConstraintOperators_DoesNotContainResourceOfSubtype(t *testing.T) {
 		})
 	}
 }
+
+func TestConstraintOperators_DoNotSendNotification(t *testing.T) {
+	ctx := context.TODO()
+	testErr := errors.New("test error")
+
+	applicationTypeLabel := "applicationType"
+	runtimeTypeLabel := "runtimeType"
+	inputAppType := "input-type"
+	inputAppID := "eb2d5110-ca3a-11ed-afa1-0242ac120002"
+	appID := "b55131c4-ca3a-11ed-afa1-0242ac120002"
+	runtimeID := "c66341c4-ca3a-11ed-afa1-0242ac120564"
+	runtimeCtxID := "f7156h4-ca3a-11ed-afa1-0242ac121237"
+
+	exceptType := "except-type"
+	formationType := "formationType"
+
+	in := &formationconstraintpkg.DoNotSendNotificationInput{
+		ResourceType:       model.ApplicationResourceType,
+		ResourceSubtype:    inputAppType,
+		ResourceID:         inputAppID,
+		SourceResourceType: model.ApplicationResourceType,
+		SourceResourceID:   appID,
+		Tenant:             testTenantID,
+		ExceptSubtypes:     []string{exceptType},
+	}
+
+	inWithFormationTypeException := &formationconstraintpkg.DoNotSendNotificationInput{
+		ResourceType:         model.ApplicationResourceType,
+		FormationTemplateID:  formationTemplateID,
+		ResourceSubtype:      inputAppType,
+		ResourceID:           inputAppID,
+		SourceResourceType:   model.ApplicationResourceType,
+		SourceResourceID:     appID,
+		Tenant:               testTenantID,
+		ExceptSubtypes:       []string{exceptType},
+		ExceptFormationTypes: []string{formationType},
+	}
+
+	runtimeIn := &formationconstraintpkg.DoNotSendNotificationInput{
+		ResourceType:       model.ApplicationResourceType,
+		ResourceSubtype:    inputAppType,
+		ResourceID:         inputAppID,
+		SourceResourceType: model.RuntimeResourceType,
+		SourceResourceID:   runtimeID,
+		Tenant:             testTenantID,
+		ExceptSubtypes:     []string{exceptType},
+	}
+
+	runtimeContextIn := &formationconstraintpkg.DoNotSendNotificationInput{
+		ResourceType:       model.ApplicationResourceType,
+		ResourceSubtype:    inputAppType,
+		ResourceID:         inputAppID,
+		SourceResourceType: model.RuntimeContextResourceType,
+		SourceResourceID:   runtimeCtxID,
+		Tenant:             testTenantID,
+		ExceptSubtypes:     []string{exceptType},
+	}
+
+	testCases := []struct {
+		Name                  string
+		Input                 formationconstraint.OperatorInput
+		LabelSvc              func() *automock.LabelService
+		RuntimeContextRepo    func() *automock.RuntimeContextRepo
+		FormationTemplateRepo func() *automock.FormationTemplateRepo
+		ExpectedResult        bool
+		ExpectedErrorMsg      string
+	}{
+		{
+			Name:  "Success for a system when notifications should be skipped",
+			Input: in,
+			LabelSvc: func() *automock.LabelService {
+				svc := &automock.LabelService{}
+				svc.On("GetByKey", ctx, testTenantID, model.ApplicationLabelableObject, appID, applicationTypeLabel).Return(&model.Label{Value: inputAppType}, nil).Once()
+				return svc
+			},
+			ExpectedResult:   false,
+			ExpectedErrorMsg: "",
```

Comment:
In `success` tests this `ExpectedErrorMsg` can be skipped, it will use the default string value which is `""`.